### PR TITLE
Lured Pokestop labels now say "Waiting for scan..." when contents are unknown

### PR DESCRIPTION
**Pull request by hgentry**
## Description

Added pokestops to stale marker checks to replace them when their data is no longer accurate
## How Has This Been Tested?

I've been observing downtown Indianapolis to make sure things are functioning. Check it out here: http://68.58.146.192:5000
## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

… unknown

### CODE REVIEW

File: static/map.js [hunk 1]
```diff
@@ -451,6 +462,16 @@ function clearStaleMarkers() {
             delete map_lure_pokemons[key];
         }
     });
+    
+    $.each(map_pokestops, function(key, value) {
+
+        if (!!!map_pokestops[key]['dead'] && !!map_pokestops[key]['lure_expiration'] && map_pokestops[key]['lure_expiration'] < new Date().getTime()) {
```
Comment (andrewda) [inline static/map.js]: `!!!` wat

Comment (hgentry) [inline static/map.js]: Haha fair enough

**Comment by andrewda** [comment]
Also, lure_expirations means nothing. It is just a suggestion of when to next check to see if the lure has expired. Every 2 minutes or so, the lure_expiration falls a few seconds behind the current time, and is promptly sent forward another two minutes or so. To get the actual time remaining, you gotta subtract the `time_updated` or whatever it's called from the current time.

**Comment by hgentry** [comment]
Wait, then where does that value come from?

**Comment by hgentry** [comment]
I think it's still good this way. After a while, we can be certain that the current encounter on the lure has moved on and the map doesn't know what's there. And through testing, this seems to at least approximate it well enough that I have yet to see evidence of this being incorrect.

**Comment by jmccorm** [comment]
http://68.58.146.5000 is not a valid URL.

**Comment by hgentry** [comment]
http://68.58.146.192:5000

File: static/map.js [hunk 2]
```diff
@@ -356,10 +364,13 @@ function setupPokestopMarker(item) {
         optimized: false,
         icon: 'static/forts/' + imagename + '.png',
     });
-
-
+    current_time = (new Date()).getTime();
+    var dead = false;
+    if((new Date(item.lure_expiration)).getTime() < current_time) {
+    	dead = true;
```
Comment (Chlodochar) [inline static/map.js]: Match the indentation of surrounding lines.

File: static/map.js [hunk 3]
```diff
@@ -451,6 +462,18 @@ function clearStaleMarkers() {
...
         }
     });
+    
+    $.each(map_pokestops, function(key, value) {
+
+        if (!map_pokestops[key]['dead'] && !!map_pokestops[key]['lure_expiration'] && map_pokestops[key]['lure_expiration'] < new Date().getTime()) {
+	        map_pokestops[key].marker.setMap(null);
+	        map_pokestops[key]['dead'] = true;
+	        if (!(localStorage.showPokestops === 'true')) {
+                	map_pokestops[key].marker = setupPokestopMarker(map_pokestops[key]);
```
Comment (Chlodochar) [inline static/map.js]: Same here.

**Comment by hgentry** [comment]
problems addressed + rebase

**Comment by hgentry** [comment]
I can rebase again on my lunch break.

**Comment by Chlodochar** [comment]
It wasn't rebased correctly. I see `>>>` in there from the conflicts. It might be easier for rebasing to squash things into one commit, so that you can solve all conflicts at once. Use `git rebase -i upstream/develop` for that.

**Comment by hgentry** [comment]
Ready @Chlodochar :)

File: static/map.js [hunk 4]
```diff
@@ -731,6 +753,7 @@ function loadRawData() {
     })
 }
 
+
```
Comment (Chlodochar) [inline static/map.js]: This change is unrelated.

File: static/map.js [hunk 5]
```diff
@@ -757,7 +780,7 @@ function processPokestops(i, item) {
     else {
         item2 = map_data.pokestops[item.pokestop_id];
         if (!!item.lure_expiration != !!item2.lure_expiration || item.active_pokemon_id != item2.active_pokemon_id) {
-            item.marker.setMap(null);
+            item2.marker.setMap(null);
```
Comment (Chlodochar) [inline static/map.js]: This change is unrelated.

File: static/map.js [hunk 6]
```diff
@@ -832,6 +855,7 @@ function processScanned(i, item) {
     if (item.scanned_id in map_data.scanned) {
         map_data.scanned[item.scanned_id].marker.setOptions({fillColor: getColorByDate(item.last_modified)});
     }
+
```
Comment (Chlodochar) [inline static/map.js]: This change is unrelated.

File: static/map.js [hunk 7]
```diff
@@ -1120,10 +1144,12 @@ $(function () {
             }
         }
     }
+
```
Comment (Chlodochar) [inline static/map.js]: This change is unrelated.

File: config/config.ini.example [hunk 8]
```diff
@@ -24,4 +24,4 @@
 
 # Webserver settings
 #host:                  # address to listen on (default 127.0.0.1)
-#port:                  # port to liste on (default 5000)
+#port:                  # port to listen on (default 5000)
```
Comment (Chlodochar) [inline config/config.ini.example]: This change is unrelated.

File: static/map.js [hunk 9]
```diff
@@ -1120,10 +1144,12 @@ $(function () {
             }
         }
     }
+    
 
     // Setup UI element interactions
     $('#gyms-switch').change(buildSwitchChangeListener(map_data, "gyms", "showGyms"));
     $('#pokemon-switch').change(buildSwitchChangeListener(map_data, "pokemons", "showPokemon"));
+    $('#pokemon-switch').change(buildSwitchChangeListener(map_data, "lure_pokemons", "showPokemon"));
```
Comment (Chlodochar) [inline static/map.js]: This seems incorrect, but I do not know what the intention is.

Comment (hgentry) [inline static/map.js]: It makes the pokemon switch work for both pokemon and lured pokemon.

File: static/map.js (hunk 5 above)
Comment (hgentry) [inline static/map.js]: It fixes a bug with the pokestop toggle, but I'll take it out

File: static/map.js (hunk 9 above)
Comment (hgentry) [inline static/map.js]: All changes made

File: static/map.js [hunk 10]
```diff
@@ -567,10 +575,13 @@ function setupPokestopMarker(item) {
         optimized: false,
         icon: 'static/forts/' + imagename + '.png',
     });
-
-
+    current_time = (new Date()).getTime();
+    var dead = false;
+    if((new Date(item.lure_expiration)).getTime() < current_time) {
```
Comment (Chlodochar) [inline static/map.js]: I think this is incorrect. The pokemon times are in UTC, new Date results in local time.

File: static/map.js [hunk 11]
```diff
@@ -425,18 +425,26 @@ function pokestopLabel(lured, last_modified, active_pokemon_id, latitude, longit
...
-                <b>Lured Pokéstop</b>
-            </div>
-            <div>
+	
+	var waitText;
+	if(dead) {
+		waitText = "<div>Lured Pokemon: Waiting for scan...</div>";
+	}
+	else {
+		waitText = `<div>
```
Comment (Chlodochar) [inline static/map.js]: Use spaces for indentation if the surrounding lines have spaces as well.

File: config/config.ini.example [hunk 12]
```diff
@@ -25,3 +25,4 @@
 # Webserver settings
 #host:                  # address to listen on (default 127.0.0.1)
 #port:                  # port to listen on (default 5000)
+
```
Comment (Chlodochar) [inline config/config.ini.example]: Unrelated change.

File: static/map.js (hunk 10 above)
Comment (hgentry) [inline static/map.js]: Right, but the local time carries a timezone flag in it, so we're still comparing the correct epoch time values. Also, this line results in correct behavior.

**Comment by hgentry** [comment]
Changes made

**Comment by hgentry** [comment]
Time for me to rebase again... Here we go

**Comment by hgentry** [comment]
Rebased again @Chlodochar

**Comment by hgentry** [comment]
Ready again

File: static/dist/js/map.js [hunk 13]
```diff
@@ -0,0 +1,1275 @@
+
```
Comment (Chlodochar) [inline static/dist/js/map.js]: `static/dist` files should be left out of the PR. This should reduce the number of times you need to rebase as well.

File: config/config.ini.example [hunk 14]
```diff
@@ -28,4 +28,4 @@
 #port:                  # port to listen on (default 5000)
 
 #Uncomment a line when you want to change its default value (Remove # at the beginning)
-#username, password, location and gmaps-key are required
\ No newline at end of file
+#username, password, location and gmaps-key are required
```
Comment (Chlodochar) [inline config/config.ini.example]: This is an unrelated change.

**Comment by AHAAAAAAA** [comment]
Ready for this, but I agree you should remove dist/ from PRs in the future. Rebase and I'll test/merge
